mlx/coverity_services.py

```diff
@@ -1,614 +1,396 @@
+        If a column name does not match the name of the `columns` property, the column can not be obtained because
+        it need the correct corresponding column key.
+        Column key `cid` is always obtained to use later in other functions.
 
-        snapshot_scope.showSelector = 'last()'
-        snapshot_scope.compareSelector = 'last()'
+        Args:
+            stream (str): Name of the stream to query
+            filters (dict): Dictionary with attribute names as keys and CSV lists of attribute values to query as values
+            column_names (list[str]): The column names
 
-        logging.info('Running Coverity query...')
-        return self.client.service.getMergedDefectsForSnapshotScope(project_id, filter_spec,
-                                                                    page_spec, snapshot_scope)
+        Returns:
+            dict: The content of the request. This has a structure like:
+                {
+                    "offset": 0,
+                    "totalRows": 2720,
+                    "columns": [list of column keys]
+                    "rows": list of [list of dictionaries {"key": <key>, "value": <value>}]
+                }
+        """
+        logging.info("Querying Coverity for defects in stream [%s] ...", stream)
+        query_filters = [
+            {
+                "columnKey": "streams",
+                "matchMode": "oneOrMoreMatch",
+                "matchers": [
+                    {
+                        "class": "Stream",
+                        "name": stream,
+                        "type": "nameMatcher"
+                    }
+                ]
+            }
+        ]
+
+        Filter = namedtuple("Filter", "name matcher_type list allow_regex", defaults=[None, False])
+        filter_options = {
+            "checker": Filter("Checker", "keyMatcher", self.checkers, True),
+            "impact": Filter("Impact", "keyMatcher", IMPACT_LIST),
+            "kind": Filter("Issue Kind", "keyMatcher", KIND_LIST),
+            "classification": Filter("Classification", "keyMatcher", CLASSIFICATION_LIST),
+            "action": Filter("Action", "keyMatcher", ACTION_LIST),
+            "cwe": Filter("CWE", "idMatcher"),
+            "cid": Filter("CID", "idMatcher")
+        }
+
+        for option, filter in filter_options.items():
+            if filters[option]:
+                filter_values = self.handle_attribute_filter(
+                    filters[option], filter.name, filter.list, filter.allow_regex
+                )
+                if filter_values:
+                    query_filters.append(self.assemble_query_filter(filter.name, filter_values, filter.matcher_type))
+
+        if filters["component"]:
+            filter_values = self.handle_component_filter(filters["component"])
+            if filter_values:
+                query_filters.append(self.assemble_query_filter("Component", filter_values, "nameMatcher"))
```

Comment:
```suggestion
        if filters["component"] and filter_values := self.handle_component_filter(filters["component"])            query_filters.append(self.assemble_query_filter("Component", filter_values, "nameMatcher"))
```
Python>=3.8 only, example of the new walrus operator

Reply:
Should I delete Python 3.7 in this PR or in a new one to test this new walrus operator?

Reply:
A separate PR based on master branch, so we can merge that one first.

Reply:
This is just one example of an assignment expression. Can you find more use cases?